packages/config/src/VitessceAutoConfig.js

```diff
@@ -0,0 +1,387 @@
+             }
+        };
+
+        this.meta["obsm"].forEach(key => {
+            if (key.toLowerCase().includes(("obsm/x_segmentations"))) {
+                options["obsSegmentations"] = {"path": key};
+            }
+            if (key.toLowerCase().includes(("obsm/x_spatial"))) {
+                options["obsLocations"] = {"path": key};
+            }
+            if (key.toLowerCase().includes("obsm/x_umap")) {
+                options["obsEmbedding"].push({"path": key,"embeddingType": "UMAP"})
+            }
+            if (key.toLowerCase().includes("obsm/x_tsne")) {
+                options["obsEmbedding"].push({"path": key,"embeddingType": "t-SNE"})
+            }
+            if (key.toLowerCase().includes("obsm/x_pca")) {
+                options["obsEmbedding"].push({"path": key,"embeddingType": "PCA"})
+            }
+        })
+    
+        this.meta["obs"].forEach(key => {
+            if (key.toLowerCase().includes("cluster") || key.toLowerCase().includes("cell_type")) {
+                if (!("obsSets" in options)) {
+                    options["obsSets"] = [
+                        {
+                        "name": "Cell Type",
+                        "path": [key]
+                        }
+                    ]
+                }
+                else {
+                    options.obsSets[0].path.push(key)
+                }
+            }
+        });
+    
+        return {
+            options: options,
+            fileType: this.fileType,
+            url: this.fileUrl,
+            coordinationValues: {
+                "obsType": "cell",
+                "featureType": "gene",
+                "featureValueType": "expression"
+            }
+        };
+    }
+
+    composeViewsConfig() {
+        
+        let views = [];
+
+        const hasCellSetData = this.meta["obs"]
+            .find(key => key.toLowerCase().includes("cluster") || key.toLowerCase().includes("cell_type"));
+
+        if (hasCellSetData.length > 0) {
+            views.push(['obsSets']);
+        }
+    
+        this.meta["obsm"].forEach(key => {
+            if (key.toLowerCase().includes("obsm/x_umap")) {
+                views.push(['scatterplot', { mapping: 'UMAP' }])
+            }
+            if (key.toLowerCase().includes("obsm/x_tsne")) {
+                views.push(['scatterplot', { mapping: 't-SNE' }])
+            }
+            if (key.toLowerCase().includes("obsm/x_pca")) {
+                views.push(['scatterplot', { mapping: 'PCA' }])
+            }
+            if (key.toLowerCase().includes(("obsm/x_segmentations"))) {
+                views.push(['layerController'])
+            }
+            if (key.toLowerCase().includes(("obsm/x_spatial"))) {
+                views.push(['spatial'])
+            }
+        })
+
+        if (this.meta["X"]) {
+            views.push(["heatmap"]);
+            views.push(["featureList"]);
+        }
+
+        return views;
+    }
+}
+
+export class VitessceAutoConfig {
```

Comment:
The usage of classes makes a lot of sense for `OmeTiffAutoConfig` and `AnndataZarrAutoConfig`. 

Could `VitessceAutoConfig` be refactored into an `async` function instead, e.g., `generateConfig(fileUrl)`? This could compose the current `calculateCoordinates`, `getFileType` , and `generateConfig`, potentially making it easier to test these as standalone functions that do not rely on `this`